parsers/IN_WE.py

```diff
@@ -18,153 +18,144 @@
     except json.decoder.JSONDecodeError:
         raise ParserException(
             parser="IN_WE.py",
-            message=f"{target_datetime}: {kind} data is not available",
+            message=f"{target_datetime}: {url} data is not available",
         )
-
-    datetime_col = KIND_MAPPING[kind]["datetime_column"]
-    for item in data:
-        item[datetime_col] = datetime.strptime(item[datetime_col], "%Y-%d-%m %H:%M:%S")
-        dt = arrow.get(item[datetime_col])
-        if dt.second >= 30:
-            item[datetime_col] = dt.shift(minutes=1).floor("minute").datetime
-        else:
-            item[datetime_col] = dt.floor("minute").datetime
     return data
 
 
-def filter_raw_data(
-    kind: str,
-    data: dict,
-    target_datetime: datetime,
-) -> pd.DataFrame:
+def process_exchanges_data(
+    data: dict, zone_key1: str, zone_key2: str, logger: Logger
+) -> ExchangeList:
     """
-    Filters out correct datetimes (source data is 12 hour format)
+    Processes exchanges data:
+    - filter out the records that correspond to the zone_key
+    - convert the datetime field to contain only date and hour(remove minute and secods)
+    - set timezone to IST for datetime field
+    - average all data points grouping by target_datetime hour
+    - generate the Exchange list from the Current_Loading column
     """
-    assert len(data) > 0
-    assert kind != ""
-
-    dt_12_hour = arrow.get(target_datetime.strftime("%Y-%m-%d %I:%M")).datetime
-    datetime_col = KIND_MAPPING[kind]["datetime_column"]
-    filtered_data = pd.DataFrame(
-        [item for item in data if item[datetime_col].hour == dt_12_hour.hour]
-    )
-    return filtered_data
-
 
-def format_exchanges_data(
-    data: dict, zone_key1: str, zone_key2: str, target_datetime: datetime
-) -> float:
-    """format exchanges data:
-    - filters out correct datetimes (source data is 12 hour format)
-    - average all data points in the target_datetime hour"""
-    assert target_datetime is not None
     assert len(data) > 0
 
-    sortedZoneKeys = "->".join(sorted([zone_key1, zone_key2]))
-    filtered_data = filter_raw_data(
-        kind="exchange", data=data, target_datetime=target_datetime
-    )
+    datetime_column = "lastUpdate"
+    value_column = "Current_Loading"
+    zone_key = "->".join(sorted([zone_key1, zone_key2]))
 
-    filtered_data["zone_key"] = filtered_data["Region_Name"].map(EXCHANGES_MAPPING)
-    df_exchanges = filtered_data.loc[filtered_data["zone_key"] == sortedZoneKeys]
+    df = pd.DataFrame(data, columns=["Region_Name", datetime_column, value_column])
+    df = df[df["Region_Name"] == EXCHANGES_MAPPING[zone_key]]
+    df[datetime_column] = (
+        pd.to_datetime(df[datetime_column], format="%Y-%m-%d %H:%M:%S")
+        .dt.tz_localize(IN_TZ)
+        .dt.floor("h")
+    )
+    df = df.groupby(["Region_Name", datetime_column]).mean().round(3)
+    df[value_column] = -df[value_column]
 
-    if target_datetime.hour >= 12:
```

Comment:
This logic was to compensate for 12 hr dataformat, we do not need it in the newer version